server/StudySharp.API/Controllers/CourseController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Threading.Tasks;
+using AutoMapper;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using StudySharp.API.Requests.Courses;
+using StudySharp.API.Responses.Courses;
+using StudySharp.ApplicationServices.Commands;
+using StudySharp.ApplicationServices.Queries;
+using StudySharp.Domain.General;
+
+namespace StudySharp.API.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("api/courses")]
+
+    public class CourseController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+        private readonly IMapper _mapper;
+
+        public CourseController(IMediator mediator, IMapper mapper)
+        {
+            _mediator = mediator;
+            _mapper = mapper;
+        }
+
+        [HttpPost]
+        public async Task<OperationResult> Add([FromBody] AddCourseRequest addCourseRequest)
+        {
+            var addCourseCommand = _mapper.Map<AddCourseCommand>(addCourseRequest);
+            return await _mediator.Send(addCourseCommand);
+        }
+
+        [HttpDelete("{id:int}")]
+        public async Task<OperationResult> Remove([FromRoute] RemoveCourseByIdRequest removeCourseByIdRequest)
+        {
+            var removeCourseByIdCommand = _mapper.Map<RemoveCourseByIdCommand>(removeCourseByIdRequest);
+            return await _mediator.Send(removeCourseByIdCommand);
+        }
+
+        [HttpGet("{id:int}")]
+        public async Task<OperationResult<GetCourseByIdResponse>> GetCourseById([FromRoute] GetCourseByIdRequest getCourseByIdRequest)
+        {
+            var getCourseByIdQuery = _mapper.Map<GetCourseByIdQuery>(getCourseByIdRequest);
+            var operationResult = await _mediator.Send(getCourseByIdQuery);
+
+            if (!operationResult.IsSucceeded)
+            {
+                return OperationResult.Fail<GetCourseByIdResponse>(operationResult.Errors);
+            }
+
+            var response = _mapper.Map<GetCourseByIdResponse>(operationResult.Result);
+            return OperationResult.Ok(response);
+        }
+
+        [HttpGet("{teacherId:int}")]
+        public async Task<OperationResult<GetCoursesByTeacherIdResponse>> GetCoursesByTeacherId([FromRoute] GetCoursesByTeacherIdRequest getCoursesByTeacherIdRequest)
```

Comment:
"Microsoft.AspNetCore.Routing.Matching.AmbiguousMatchException: The request matched multiple endpoints. Matches: \r\n\r\nStudySharp.API.Controllers.CourseController.GetCourseById (StudySharp.API)\r\nStudySharp.API.Controllers.CourseController.GetCoursesByTeacherId (StudySharp.API)

Reply:
1. HttpGet("{teacherId:int}")] => [HttpGet] 
2. add [Route("~/api/teachers/{teacherId:int}/courses")] at the top of the method

And make sure everything works fine because I see that it was not tested locally